src/RichText/replaceLink.tsx

```diff
@@ -1,10 +1,11 @@
 import type {Element} from 'domhandler';
 import type {DOMNode} from 'html-react-parser';
-import {domToReact} from 'html-react-parser';
 import type {LinkComponent, ImageComponent, MacroComponent, Replacer, RichTextData, LinkComponentParams} from '../types';
 import type {createReplacer as CreateReplacer} from './createReplacer';
+
+import * as htmlReactParser from 'html-react-parser';
```

Comment:
I suspect it had to do with build system in starter-react4xp when using tsconfig paths to point to local files rather than node module.

Whether `import {domToReact} from 'html-react-parser'` works depends on what allowSyntheticDefaultImports and esModuleInterop are set to, while `import * as htmlReactParser from 'html-react-parser'` should work regardless.